components/sidebar/sidebar_menu/sidebar_menu.tsx

```diff
@@ -39,21 +44,44 @@ export default class SidebarMenu extends React.PureComponent<Props, State> {
             width: 0,
         };
 
+        this.menuWrapperRef = React.createRef();
         this.menuButtonRef = React.createRef();
     }
 
+    componentDidUpdate(prevProps: Props) {
+        if (prevProps.draggingState.state !== this.props.draggingState.state && this.props.draggingState.state === DraggingStates.CAPTURE) {
+            this.closeMenu();
+        }
+    }
+
     // TODO: Temporary code to keep the menu in place while scrolling
+    // This shouldn't be necessary once the menus are fixed up
     componentDidMount() {
         const scrollbars = document.querySelectorAll('#SidebarContainer .SidebarNavContainer .scrollbar--view');
         if (scrollbars && scrollbars[0]) {
-            scrollbars[0].addEventListener('scroll', this.setMenuPosition);
+            scrollbars[0].addEventListener('scroll', this.closeMenu);
         }
     }
 
     componentWillUnmount() {
         const scrollbars = document.querySelectorAll('#SidebarContainer .SidebarNavContainer .scrollbar--view');
         if (scrollbars && scrollbars[0]) {
-            scrollbars[0].removeEventListener('scroll', this.setMenuPosition);
+            scrollbars[0].removeEventListener('scroll', this.closeMenu);
+        }
+    }
+
+    closeMenu = () => {
+        if (this.menuWrapperRef.current) {
+            this.menuWrapperRef.current.setState({open: false});
+        }
+        this.handleMenuToggle(false);
+    }
+
+    // Set the z-index on the sidebar scrollbar while a menu is open so that it doesn't float on top of menus
+    disableScrollbar = () => {
+        const scrollbars: NodeListOf<HTMLElement> = document.querySelectorAll('#SidebarContainer .SidebarNavContainer .scrollbar--view');
```

Comment:
Doesn't work. `z-index` is determined first and foremost by the parent element. The scrollbars sit outside of the entire container, and the menus are rendered within the container, thus since the scrollbars need to display on top of the container, their `z-index` always needs to be higher. 

I suppose I could do this to the container instead of the scrollbars, but it's effectively the same procedure. This is definitely a hack and is necessary only because the menu component is not hoisted, which would fix this issue.